tests/end_to_end_test/repository_connect_test.go

```diff
@@ -77,6 +79,19 @@ func TestReconnect(t *testing.T) {
 	e.RunAndExpectSuccess(t, "repo", "status")
 }
 
+func TestReconnectWithDoNotWaitForUpgrade(t *testing.T) {
+	t.Parallel()
+
+	runner := testenv.NewInProcRunner(t)
+	e := testenv.NewCLITest(t, testenv.RepoFormatNotImportant, runner)
+
+	e.RunAndExpectSuccess(t, "repo", "create", "filesystem", "--path", e.RepoDir)
+	e.RunAndExpectSuccess(t, "repo", "disconnect")
+	filesystem.New(context.Background(), &filesystem.Options{Path: e.RepoDir}, false)
+	e.RunAndExpectSuccess(t, "repo", "connect", "filesystem", "--path", e.RepoDir)
+	e.RunAndExpectSuccess(t, "repo", "status")
```

Comment:
I think we need to cover the targeted scenario as well, no?
```suggestion
  e := testenv.NewCLITest(t, testenv.RepoFormatNotImportant, runner)
  e.Environment["KOPIA_UPGRADE_LOCK_ENABLED"] = "1"

	e.RunAndExpectSuccess(t, "repo", "create", "filesystem", "--path", e.RepoDir, "--format-version", "2")
	e.RunAndExpectSuccess(t, "repo", "upgrade", "begin", "--lock-only", "--upgrade-owner-id", "admin")
	e.RunAndExpectSuccess(t, "repo", "disconnect")
	e.RunAndExpectFailure(t, "repo", "connect", "filesystem", "--path", e.RepoDir, "--upgrade-no-block")
```